packages/driver/src/cy/commands/xhr.js

```diff
@@ -0,0 +1,533 @@
+  waitOnResponses: undefined,
+  onAbort: undefined,
+  onRequest: undefined, // need to rebind these to 'cy' context
+  onResponse: undefined,
+}
+
+module.exports = (Commands, Cypress, cy, state, config) => {
+  reset()
+
+  // if our page is going away due to
+  // a form submit / anchor click then
+  // we need to cancel all pending
+  // XHR's so the command log displays
+  // correctly
+  Cypress.on('window:unload', cancelPendingXhrs)
+
+  Cypress.on('test:before:run', () => {
+    // reset the existing server
+    reset()
+
+    // create the server before each test run
+    // its possible for this to fail if the
+    // last test we ran ended with an invalid
+    // window such as if the last test ended
+    // with a cross origin window
+    try {
+      server = startXhrServer(cy, state, config)
+    } catch (err) {
+      // in this case, just don't bind to the server
+      server = null
+    }
+
+    return null
+  })
+
+  Cypress.on('window:before:load', (contentWindow) => {
+    if (server) {
+      // dynamically bind the server to whatever is currently running
+      return server.bindTo(contentWindow)
+    }
+
+    server = startXhrServer(cy, state, config)
+  })
+
+  return Commands.addAll({
+    server (options) {
+      let userOptions = options
+
+      if (arguments.length === 0) {
+        userOptions = {}
+      }
+
+      if (!_.isObject(userOptions)) {
+        $errUtils.throwErrByPath('server.invalid_argument')
+      }
+
+      options = _.defaults({}, userOptions, {
+        enable: true, // set enable to false to turn off stubbing
+      })
+
+      // if we disable the server later make sure
+      // we cannot add cy.routes to it
+      state('serverIsStubbed', options.enable)
+
+      return getXhrServer(state).set(options)
+    },
+
+    route (...args) {
+      // TODO:
+      // if we return a function which returns a promise
+      // then we should be handling potential timeout issues
+      // just like cy.then does
+
+      // method / url / response / options
+      // url / response / options
+      // options
+
+      // by default assume we have a specified
+      // response from the user
+      let hasResponse = true
+
+      if (!state('serverIsStubbed')) {
+        $errUtils.throwErrByPath('route.failed_prerequisites')
+      }
+
+      // get the default options currently set
+      // on our server
+      let o = getXhrServer(state).getOptions()
+      let options = o
+
+      // enable the entire routing definition to be a function
+      const parseArgs = (...args) => {
+        if (_.isObject(args[0]) && !_.isRegExp(args[0])) {
+          // we dont have a specified response
+          if (!_.has(args[0], 'response')) {
+            hasResponse = false
+          }
+
+          options = (o = _.extend({}, options, args[0]))
+        } else if (args.length === 0) {
+          $errUtils.throwErrByPath('route.invalid_arguments')
+        } else if (args.length === 1) {
```

Comment:
these following `else if` with `arge.length === n` exclude each other, so can just be standard `if` statements. Always prefer standalone `if` statements over long `else if` chains. Can you update?

Reply:
It should not be updated. Because when executing `if-else` statement, when the first condition met, other conditions below are ignored. 

In this case, if we remove all `else`s, when the first argument is object type, the first condition and `args.length === 1` are met. So, it breaks the program. [Actually, it did so.](https://circleci.com/gh/cypress-io/cypress/312191?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)